From 85174a4ade0dfa033290bf0be474480161b1bc7e Mon Sep 17 00:00:00 2001
From: "Zheng, Lei" <realthunder.dev@gmail.com>
Date: Tue, 23 Jun 2020 10:37:50 +0800
Subject: [PATCH] Fix transparent SoShape losing per line/point color

When rendering with ShapeStyleElement::transparencyType of
SORTED_OBJECT_SORTED_TRIANGLE_BLEND
---
 src/shapenodes/SoShape.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/shapenodes/SoShape.cpp b/src/shapenodes/SoShape.cpp
index 0e0ffcc2f..382fd7a36 100644
--- a/src/shapenodes/SoShape.cpp
+++ b/src/shapenodes/SoShape.cpp
@@ -611,7 +611,7 @@ SoShape::shouldGLRender(SoGLRenderAction * action)
       if (nelem->getNum() == 0) {
         glPushAttrib(GL_LIGHTING_BIT);
         glDisable(GL_LIGHTING);
-        arrays &= SoPrimitiveVertexCache::NORMAL;
+        arrays &= ~SoPrimitiveVertexCache::NORMAL;
       }
       PRIVATE(this)->pvcache->renderLines(state, arrays);
       PRIVATE(this)->pvcache->renderPoints(state, arrays);
-- 
2.23.0

